check_limits.py

```diff
@@ -1,18 +1,36 @@
+battery_limits = {"temperature": [0, 45], "soc": [20, 80], "charge_rate": [0, 0.8]}
+					
+warn_limit = 5
 
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
+lang_msg ={'DE' : {'warning' : "Warnung! Bitte halten Sie innerhalb des Bereichs",'Okay' : "Batterie ist in einwandfreiem Zustand", 'fail' : "Batterielimits A-Range!"},
+	   'EN' : {'warning' : "Warning! Please maintain wihtin Range", 'Okay' : "Battery in Perfect Condition", 'fail' : " Battery Limits out of Range!"}}
 
-  return True
+lang = "EN"
 
+def battery_is_ok(battery_values):
+	value_range = battery_limits[battery_values["battery_parameter"]]
+	low = value_range[0]
+	high = value_range[1]
+	warning_value = (high * warn_limit) / 100
+	low_warning = low + warning_value
+	high_warning = high - warning_value
+	value = battery_values["value"]
+	if value <= low or value >= high:
+		print(lang_msg[lang]['fail'])
```

Comment:
Can we separate output from processing? What if the print needs to change and send something over the network instead?